src/Mod/Draft/draftobjects/draft_annotation.py

```diff
@@ -42,6 +42,18 @@ def __init__(self, obj, tp="Annotation"):
         self.Type = tp
 
 
+    def onDocumentRestored(self, obj):
+        if hasattr(obj, "ViewObject"):
```

Comment:
if this is run on console mode (no GUI), the obj will have a "ViewObject" attribute, but obj.ViewObject  is None. So the next line will fail as None has no PropertiesList attribute. so I think the above line should be:
`if hasattr(obj, "ViewObject") and obj.ViewObject:`

Reply:
thanks, keep learning :)

Reply:
actually I really prefer the way you did in Arch: a separated `set_properties()` method that is called `onDocumentRestored` and checks if all the properties are present...